client/pages/create.jsx

```diff
@@ -72,95 +89,106 @@ export default class Create extends React.Component {
       startTime: '',
       endTime: '',
       location: '',
-      status: false
+      isOpen: false
     });
+    this.props.getPosts();
```

Comment:
The reason it doesn't update properly sometimes is because you don't want to call `getPosts` until after the POST request has completed, also since you are already on the home page you don't need to change the location anymore. Update your `handleSubmit` like:

```JSX
  handleSubmit(props) {
    event.preventDefault();
    fetch('/api/posts', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(this.state)
    })
      .then(response => response.json())
      .then(() => {
        // By moving the state update and call to getPosts to inside
        // the .then callback, this stuff wont happen until the request has finished
        // which is what you want
        this.setState({
          post: '',
          price: '',
          startDate: '',
          startTime: '',
          endTime: '',
          location: '',
          status: false
          isOpen: false
        });
        this.props.getPosts();
      });
```